File: epi_judge_java/epi/DeleteKthLastFromList.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class DeleteKthLastFromList {

    @EpiTest(testDataFile = "delete_kth_last_from_list.tsv")
    // Assumes L has at least k nodes, deletes the k-th last node in L.
    public static ListNode<Integer> removeKthLast(ListNode<Integer> head, int k) {
        var dummy = new ListNode<Integer>(-1, head);
        var runner = dummy;
        for (int i = 0; i < k + 1; i++) {
            runner = runner.next;
        }
        var toRemovePrev = dummy;
        while (runner != null) {
            toRemovePrev = toRemovePrev.next;
            runner = runner.next;
        }
        toRemovePrev.next = toRemovePrev.next.next;
        return dummy.next;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "DeleteKthLastFromList.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/delete_kth_last_from_list.cc
#include <memory>

#include "list_node.h"
#include "test_framework/generic_test.h"

using std::shared_ptr;

// Assumes L has at least k nodes, deletes the k-th last node in L.
shared_ptr<ListNode<int>> RemoveKthLast(const shared_ptr<ListNode<int>> &head, int k) {
    auto first = head;
    for (int i = 0; i < k; ++i) {
        first = first->next;
    }

    if (!first) {
        return head->next;
    }

    first = first->next;

    auto second = head;
    while (first) {
        first = first->next;
        second = second->next;
    }

    second->next = second->next->next;

    return head;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"L", "k"};
    return GenericTestMain(args, "delete_kth_last_from_list.cc",
                           "delete_kth_last_from_list.tsv", &RemoveKthLast,
                           DefaultComparator{}, param_names);
}
